riposte-core/src/test/java/com/nike/riposte/server/handler/RoutingHandlerTest.java

```diff
@@ -182,4 +196,99 @@ public void findSingleEndpointForExecution_throws_MultipleMatchingEndpointsExcep
         // when
         handlerSpy.findSingleEndpointForExecution(requestInfoMock);
     }
+
+    @Test
+    public void doChannelRead_HttpRequest_throws_exception_when_content_length_header_greater_than_configured_global_request_limit() {
+        // given
+        doReturn(null).when(endpointMock).maxRequestSizeInBytesOverride();
+
+        maxRequestSizeInBytes = 10;
+        httpHeaders.set(CONTENT_LENGTH, 100);
+        httpHeaders.set(CONTENT_TYPE, APPLICATION_JSON);
+        handlerSpy = spy(new RoutingHandler(endpoints, maxRequestSizeInBytes));
+
+        // when
+        Throwable thrownException = Assertions.catchThrowable(() -> handlerSpy.doChannelRead(ctxMock, msg));
+
+        // then
+        assertThat(thrownException).isExactlyInstanceOf(TooLongFrameException.class);
+        assertThat(thrownException.getMessage()).isEqualTo("Content-Length header value exceeded configured max request size of 10");
+    }
+
+    @Test
+    public void doChannelRead_HttpRequest_endpoint_overridden_max_request_size_throws_exception() {
+        // given
+        doReturn(100).when(endpointMock).maxRequestSizeInBytesOverride();
+
+        maxRequestSizeInBytes = 99;
+        httpHeaders.set(CONTENT_LENGTH, 101);
+        httpHeaders.set(CONTENT_TYPE, APPLICATION_JSON);
+
+        handlerSpy = spy(new RoutingHandler(endpoints, maxRequestSizeInBytes));
+
+        // when
+        Throwable thrownException = Assertions.catchThrowable(() -> handlerSpy.doChannelRead(ctxMock, msg));
+
+        // then
+        assertThat(thrownException).isExactlyInstanceOf(TooLongFrameException.class);
+        assertThat(thrownException.getMessage()).isEqualTo("Content-Length header value exceeded configured max request size of 100");
+    }
+
+    @Test
+    public void doChannelRead_HttpRequest_under_max_global_request_size_processed_successfully() {
+        // given
+        doReturn(null).when(endpointMock).maxRequestSizeInBytesOverride();
+
+        maxRequestSizeInBytes = 101;
+        httpHeaders.set(CONTENT_LENGTH, 100);
+        httpHeaders.set(CONTENT_TYPE, APPLICATION_JSON);
+        handlerSpy = spy(new RoutingHandler(endpoints, maxRequestSizeInBytes));
+
+        // when
+        PipelineContinuationBehavior result = handlerSpy.doChannelRead(ctxMock, msg);
+
+        // then
+        assertThat(result).isEqualTo(PipelineContinuationBehavior.CONTINUE);
+    }
+
+    @Test
+    public void doChannelRead_HttpRequest_transfer_encoding_chunked_processed() {
```

Comment:
Confusing test name. Recommend: `doChannelRead_HttpRequest_does_not_throw_TooLongFrameException_if_content_length_header_is_missing()`. I'd also suggest making it a `@DataProvider` test with a `boolean isChunkedTransferEncoding` argument to test when transfer-encoding is chunked or when all relevant headers are just missing.